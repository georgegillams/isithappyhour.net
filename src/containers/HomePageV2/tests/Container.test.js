import React from 'react';
import { render } from '@testing-library/react';

import Home from '../Container';

describe('<Home />', () => {
  it('should render correctly', () => {
    const { container } = render(<Home authenticatorState={{ user: undefined }} />);

    expect(container).toMatchSnapshot();
  });

  it('should render correctly with no user', () => {
    const { container } = render(<Home authenticatorState={{ user: null }} />);

    expect(container).toMatchSnapshot();
  });

  it('should render correctly with user', () => {
    const { container } = render(<Home authenticatorState={{ user: {} }} />);

    expect(container).toMatchSnapshot();
  });

  it('should render correctly with admin user', () => {
    const { container } = render(<Home authenticatorState={{ user: { admin: true } }} />);

    expect(container).toMatchSnapshot();
  });
});
