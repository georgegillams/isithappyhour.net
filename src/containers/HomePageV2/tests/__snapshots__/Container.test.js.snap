// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render correctly 1`] = `
<div>
  <div
    class="sc-bczRLJ gmscGQ page-containerstyles__StyledPageContainer-sc-19l7k4-0"
    width="default"
  >
    <div
      class="sc-ftvSup laBHLg"
    >
      <style>
        .js .animated-content__wrapper--hidden {
  padding-top: 4rem;
  opacity: 0.01;
}
      </style>
      <div
        class="sc-iBkjds bhKksC animated-content__wrapper--hidden"
      >
        <div
          class="homestyles__AnswerWrapper-sc-d0smz9-0 iWmeiL"
        >
          <span
            class="sc-jSMfEi cSTHCK homestyles__Answer-sc-d0smz9-1 kSqXIm"
          >
            YES!
          </span>
        </div>
      </div>
    </div>
    <div
      class="sc-ftvSup laBHLg"
    >
      <style>
        .js .animated-content__wrapper--hidden {
  padding-top: 4rem;
  opacity: 0.01;
}
      </style>
      <div
        class="sc-iBkjds bhKksC animated-content__wrapper--hidden"
      >
        <span
          class="sc-kDDrLX bMvsic"
        >
          <span
            class="sc-jSMfEi kskSGE"
          >
            In undefined it's undefined:undefined, so it's likely to be happy hour there.
          </span>
        </span>
      </div>
    </div>
    <div
      class="sc-ftvSup laBHLg"
    >
      <style>
        .js .animated-content__wrapper--hidden {
  padding-top: 4rem;
  opacity: 0.01;
}
      </style>
      <div
        class="sc-iBkjds bhKksC animated-content__wrapper--hidden"
      >
        <span
          class="sc-kDDrLX bMvsic"
        >
          <span
            class="sc-jSMfEi kskSGE"
          >
            Travelling all that way would be environmentally unfriendly, so you're really doing us all a favour by staying where you are and having a beer.
          </span>
        </span>
      </div>
    </div>
    <div
      class="sc-ftvSup laBHLg"
    >
      <style>
        .js .animated-content__wrapper--hidden {
  padding-top: 4rem;
  opacity: 0.01;
}
      </style>
      <div
        class="sc-iBkjds bhKksC animated-content__wrapper--hidden"
      >
        <span
          class="sc-kDDrLX bMvsic"
        >
          <span
            class="sc-jSMfEi kskSGE"
          >
            Cheers üçª
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Home /> should render correctly with admin user 1`] = `
<div>
  <div
    class="sc-bczRLJ gmscGQ page-containerstyles__StyledPageContainer-sc-19l7k4-0"
    width="default"
  >
    <div
      class="sc-ftvSup laBHLg"
    >
      <style>
        .js .animated-content__wrapper--hidden {
  padding-top: 4rem;
  opacity: 0.01;
}
      </style>
      <div
        class="sc-iBkjds bhKksC animated-content__wrapper--hidden"
      >
        <div
          class="homestyles__AnswerWrapper-sc-d0smz9-0 iWmeiL"
        >
          <span
            class="sc-jSMfEi cSTHCK homestyles__Answer-sc-d0smz9-1 kSqXIm"
          >
            YES!
          </span>
        </div>
      </div>
    </div>
    <div
      class="sc-ftvSup laBHLg"
    >
      <style>
        .js .animated-content__wrapper--hidden {
  padding-top: 4rem;
  opacity: 0.01;
}
      </style>
      <div
        class="sc-iBkjds bhKksC animated-content__wrapper--hidden"
      >
        <span
          class="sc-kDDrLX bMvsic"
        >
          <span
            class="sc-jSMfEi kskSGE"
          >
            In undefined it's undefined:undefined, so it's likely to be happy hour there.
          </span>
        </span>
      </div>
    </div>
    <div
      class="sc-ftvSup laBHLg"
    >
      <style>
        .js .animated-content__wrapper--hidden {
  padding-top: 4rem;
  opacity: 0.01;
}
      </style>
      <div
        class="sc-iBkjds bhKksC animated-content__wrapper--hidden"
      >
        <span
          class="sc-kDDrLX bMvsic"
        >
          <span
            class="sc-jSMfEi kskSGE"
          >
            Travelling all that way would be environmentally unfriendly, so you're really doing us all a favour by staying where you are and having a beer.
          </span>
        </span>
      </div>
    </div>
    <div
      class="sc-ftvSup laBHLg"
    >
      <style>
        .js .animated-content__wrapper--hidden {
  padding-top: 4rem;
  opacity: 0.01;
}
      </style>
      <div
        class="sc-iBkjds bhKksC animated-content__wrapper--hidden"
      >
        <span
          class="sc-kDDrLX bMvsic"
        >
          <span
            class="sc-jSMfEi kskSGE"
          >
            Cheers üçª
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Home /> should render correctly with no user 1`] = `
<div>
  <div
    class="sc-bczRLJ gmscGQ page-containerstyles__StyledPageContainer-sc-19l7k4-0"
    width="default"
  >
    <div
      class="sc-ftvSup laBHLg"
    >
      <style>
        .js .animated-content__wrapper--hidden {
  padding-top: 4rem;
  opacity: 0.01;
}
      </style>
      <div
        class="sc-iBkjds bhKksC animated-content__wrapper--hidden"
      >
        <div
          class="homestyles__AnswerWrapper-sc-d0smz9-0 iWmeiL"
        >
          <span
            class="sc-jSMfEi cSTHCK homestyles__Answer-sc-d0smz9-1 kSqXIm"
          >
            YES!
          </span>
        </div>
      </div>
    </div>
    <div
      class="sc-ftvSup laBHLg"
    >
      <style>
        .js .animated-content__wrapper--hidden {
  padding-top: 4rem;
  opacity: 0.01;
}
      </style>
      <div
        class="sc-iBkjds bhKksC animated-content__wrapper--hidden"
      >
        <span
          class="sc-kDDrLX bMvsic"
        >
          <span
            class="sc-jSMfEi kskSGE"
          >
            In undefined it's undefined:undefined, so it's likely to be happy hour there.
          </span>
        </span>
      </div>
    </div>
    <div
      class="sc-ftvSup laBHLg"
    >
      <style>
        .js .animated-content__wrapper--hidden {
  padding-top: 4rem;
  opacity: 0.01;
}
      </style>
      <div
        class="sc-iBkjds bhKksC animated-content__wrapper--hidden"
      >
        <span
          class="sc-kDDrLX bMvsic"
        >
          <span
            class="sc-jSMfEi kskSGE"
          >
            Travelling all that way would be environmentally unfriendly, so you're really doing us all a favour by staying where you are and having a beer.
          </span>
        </span>
      </div>
    </div>
    <div
      class="sc-ftvSup laBHLg"
    >
      <style>
        .js .animated-content__wrapper--hidden {
  padding-top: 4rem;
  opacity: 0.01;
}
      </style>
      <div
        class="sc-iBkjds bhKksC animated-content__wrapper--hidden"
      >
        <span
          class="sc-kDDrLX bMvsic"
        >
          <span
            class="sc-jSMfEi kskSGE"
          >
            Cheers üçª
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Home /> should render correctly with user 1`] = `
<div>
  <div
    class="sc-bczRLJ gmscGQ page-containerstyles__StyledPageContainer-sc-19l7k4-0"
    width="default"
  >
    <div
      class="sc-ftvSup laBHLg"
    >
      <style>
        .js .animated-content__wrapper--hidden {
  padding-top: 4rem;
  opacity: 0.01;
}
      </style>
      <div
        class="sc-iBkjds bhKksC animated-content__wrapper--hidden"
      >
        <div
          class="homestyles__AnswerWrapper-sc-d0smz9-0 iWmeiL"
        >
          <span
            class="sc-jSMfEi cSTHCK homestyles__Answer-sc-d0smz9-1 kSqXIm"
          >
            YES!
          </span>
        </div>
      </div>
    </div>
    <div
      class="sc-ftvSup laBHLg"
    >
      <style>
        .js .animated-content__wrapper--hidden {
  padding-top: 4rem;
  opacity: 0.01;
}
      </style>
      <div
        class="sc-iBkjds bhKksC animated-content__wrapper--hidden"
      >
        <span
          class="sc-kDDrLX bMvsic"
        >
          <span
            class="sc-jSMfEi kskSGE"
          >
            In undefined it's undefined:undefined, so it's likely to be happy hour there.
          </span>
        </span>
      </div>
    </div>
    <div
      class="sc-ftvSup laBHLg"
    >
      <style>
        .js .animated-content__wrapper--hidden {
  padding-top: 4rem;
  opacity: 0.01;
}
      </style>
      <div
        class="sc-iBkjds bhKksC animated-content__wrapper--hidden"
      >
        <span
          class="sc-kDDrLX bMvsic"
        >
          <span
            class="sc-jSMfEi kskSGE"
          >
            Travelling all that way would be environmentally unfriendly, so you're really doing us all a favour by staying where you are and having a beer.
          </span>
        </span>
      </div>
    </div>
    <div
      class="sc-ftvSup laBHLg"
    >
      <style>
        .js .animated-content__wrapper--hidden {
  padding-top: 4rem;
  opacity: 0.01;
}
      </style>
      <div
        class="sc-iBkjds bhKksC animated-content__wrapper--hidden"
      >
        <span
          class="sc-kDDrLX bMvsic"
        >
          <span
            class="sc-jSMfEi kskSGE"
          >
            Cheers üçª
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;
